core/controllers/learner_goals.py

```diff
@@ -16,18 +16,47 @@
 
 from __future__ import annotations
 
+from core import feconf
+
 from core.constants import constants
 from core.controllers import acl_decorators
 from core.controllers import base
 from core.domain import learner_goals_services
 from core.domain import learner_progress_services
 
+from typing import Dict
+
 
-class LearnerGoalsHandler(base.BaseHandler):
+class LearnerGoalsHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
     """Handles operations related to the learner goals."""
 
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'topic_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': r'^[a-zA-Z0-9\-_]{1,12}$'
+                }]
+            }
+        },
+        'activity_type': {
+            'schema': {
+                'type': 'basestring'
+            }
+        }
```

Comment:
Should a list of choices be only allowed here?

Reply:
I intentionally not used `choices` here, because:
1. As far as I have observed, I think `activity_type` can only have `constants.ACTIVITY_TYPE_LEARN_TOPIC` as an available choice.
2. If we use `choices` here then we have to remove the else clause's exception https://github.com/oppia/oppia/blob/934e23416ebb513935c124ed2be93e26a2632785/core/controllers/learner_goals.py#L38 and if we remove this exception then backend pending line coverage error pop-up, to fix that we have to remove if clause ( which is ok, because at this time activity_type have only one choice ).
3. after removing the if clause, the `activity_type` variable is not used -- so pylint will throw an error.

Please let me know if I'm wrong somewhere or if maybe I missed something!

Reply:
1. Then only restrict it to one thing for now.
2. I think you can use assert to overcome the backend coverage
3. I think that is fine.

Basically lets only use the stuff we actually do.

Reply:
Done, added choices field in schemas